static/js/tabs.ts

```diff
@@ -85,23 +85,26 @@ function resetWindow() {
       theGlobalEditor?.setValue (adventures[tabName].loaded_program!.code);
     }
     // If there's no loaded program (either requested by id or associated to the adventure/level), load defaults.
-    else if (tabName === 'level' && window.State.default_program_name && window.State.default_program) {
+    else if (window.State.default_program_name && window.State.default_program) {
       $ ('#program_name').val(window.State.default_program_name);
       theGlobalEditor?.setValue(window.State.default_program);
     }
     else {
       if (tab.hasClass('teacher_tab')) {
-        // Todo: TB -> We should re-write the adventures structure so Teacher adventures are included in "adventures"
         $ ('#program_name').val (tabName);
         window.State.adventure_name = tabName;
         theGlobalEditor?.setValue ("");
       } else {
-        $('#program_name').val(adventures [tabName].default_save_name + ' - ' + window.State.level_title + ' ' + window.State.level);
+        if (adventures[tabName].default_save_name == 'intro') {
+          $('#program_name').val(window.State.level_title + ' ' + window.State.level);
+        } else {
+          $('#program_name').val(adventures [tabName].default_save_name + ' - ' + window.State.level_title + ' ' + window.State.level);
+        }
         theGlobalEditor?.setValue(adventures [tabName].start_code);
       }
     }
 
-    window.State.adventure_name = tabName === 'level' ? undefined : tabName;
+    window.State.adventure_name = tabName === 'intro' ? undefined : tabName;
```

Comment:
Ah indeed here it is intro and not default

Reply:
I think this is due to the dict key being called `default` but the default_save_name being `intro`. Agree that this is confusing, will take a look what would be the best approach to make this clearer!